### PR TITLE
Fix build warnings

### PR DESCRIPTION
**Pull request by sspintel**
- Replace some deprecated ngraph code usage
- Remove warning for redefined macro
- Remove warning for improper strncpy usage
- Remove unused functions and variables
- Use suggested replacements for deprecated attributes and functions in ov:: namespace

### CODE REVIEW

File: openvino_tensorflow/ovtf_builder.cc [hunk 1]
```diff
@@ -4017,13 +4017,6 @@ Status Builder::TranslateGraph(
     passes.run_passes(ng_function);
   }
   OVTF_VLOG(5) << "Done with passes";
-  //
-  // Request row-major layout on results.
-  //
-  for (auto result : ng_function->get_results()) {
-    result->set_needs_default_layout(true);
```
Comment (cavusmustafa) [inline openvino_tensorflow/ovtf_builder.cc:4024]: Why are we removing this?

**Review by cavusmustafa — APPROVED** [review]
LGTM. I just add a question in the comments.

File: openvino_tensorflow/ovtf_builder.cc (hunk 1 above)
Comment (sspintel) [reply openvino_tensorflow/ovtf_builder.cc:4024]: According to this, there's no implementation for this method anymore: https://github.com/openvinotoolkit/openvino/blob/a79830cb55623ce2654f0685b0ca51b3f412195a/src/core/include/openvino/op/result.hpp#L34

Earlier for 2021.4.2, it used to be under ngraph headers https://github.com/openvinotoolkit/openvino/blob/9424272cce8f1169c779d21a4384ac8a73e8b0c2/ngraph/core/include/ngraph/op/result.hpp#L35

File: openvino_tensorflow/backend_manager.cc [hunk 2]
```diff
@@ -78,12 +78,9 @@ Status BackendManager::GetBackendName(string& backend_name) {
 Status BackendManager::CreateBackend(shared_ptr<Backend>& backend,
                                      string& backend_name) {
   const char* env = std::getenv("OPENVINO_TF_BACKEND");
-  // Checkmarx fix. Array of max length MYRIAD.
-  char backendName[6];
 
   if (env != nullptr) {
-    strncpy((char*)backendName, env, sizeof(backendName));
```
Comment (ck-intel) [inline openvino_tensorflow/backend_manager.cc:85]: we need to find an alternative to strncpy function to fix the wrning.

**Pull request merged**